src/draft/NSG/Agent/LegalEntity/NonListedCompany/SignatoryRights.py

```diff
@@ -0,0 +1,424 @@
+    locator_name: Optional[str] = Field(
+        None,
+        title="Locator name",
+        description="Proper noun(s) applied to the real world entity identified by the locator. The locator name "
+        "could be the name of the property or complex, of the building or part of the building, "
+        "or it could be the name of a room inside a building. The key difference between a locator "
+        "designator and a locator name is that the latter is a proper name and is unlikely to include "
+        "digits.",
+        example="Shumann, Berlaymont building",
+        max_length=40,
+    )
+    address_area: Optional[str] = Field(
+        None,
+        title="Address area",
+        description="The name of a geographic area that groups Addresses. This would typically be part of a city, "
+        "a neighbourhood or village. Address area is not an administrative unit.",
+        example="Montmartre (in Paris)",
+        max_length=40,
+    )
+    post_code: Optional[str] = Field(
+        None,
+        title="Post code",
+        description="The code created and maintained for postal purposes to identify a subdivision of addresses and "
+        "postal delivery points.",
+        example="75000",
+        max_length=10,
+    )
+    post_name: Optional[str] = Field(
+        None,
+        title="Post name",
+        description="A name created and maintained for postal purposes to identify a subdivision of addresses and "
+        "postal delivery points. Usually a city.",
+        example="Paris",
+        max_length=40,
+    )
+    po_box: Optional[str] = Field(
+        None,
+        title="PO box",
+        description="A location designator for a postal delivery point at a post office, usually a number.",
+        example="9383",
+        max_length=10,
+    )
+    admin_unit_level_1: Optional[ISO_3166_1_Alpha_3] = Field(
+        None,
+        title="Admin unit level 1",
+        description="The name of the uppermost level of the address, almost always a country. ISO 3166 three "
+        "character (Alpha 3) format.",
+        example=ISO_3166_1_Alpha_3.USA,
+    )
+    admin_unit_level_2: Optional[str] = Field(
+        None,
+        title="Admin unit level 2",
+        description="The name of a secondary level/region of the address, usually a county, state or other such area "
+        "that typically encompasses several localities. Values could be a region or province, "
+        "more granular than level 1.",
+        example="Lapland",
+        max_length=40,
+    )
+
+
+class SignatoryRightsRequest(CamelCaseModel):
+    national_identifier: str = Field(
+        ...,
+        title="National identifier",
+        description="The national identifier of the non-listed company issued by the trade register",
+        example="FIN: 2464491-9 / SWE: 5560125791 / NOR: 923609016",
+        max_length=40,
+    )
+
+
+class SignatoryRightsResponse(CamelCaseModel):
+    signing_rights: List[SigningRights] = Field(
```

Comment:
@hsuomi: I think it's a bit confusing that the definition is called SigantoryRights, but we then have this field called `signingRights` containing `SigningRights` (not `signatory`).

Reply:
Agreed! I changed the class name in the spreadsheet and should be changed here as well.

Reply:
@hsuomi: Seems like the field and class were left as `signing_rights`/`SigningRights` despite the change in the spreadsheet. I'll submit a separate PR to fix this.

Reply:
Seems like I was slightly mistaken; the class was indeed renamed, but the field name was not, for some reason GitHub decided to show me the outdated version of the code when checking the comment. Might be it was because the line was changed (due to renamed class) after the original comment was created. In any case, there's now a new PR: https://github.com/Virtual-Finland/definitions/pull/46